core/src/main/java/org/bitcoinj/wallet/SendRequest.java

```diff
@@ -237,6 +238,19 @@ public static SendRequest forSubTxResetKey(NetworkParameters params, SubTxResetK
         return req;
     }
 
+    /**
+     * <p>Creates a new SendRequest for a SubTxTransition (State Transition) with credits.</p>
+     * The SubTxTransition payload must already be signed.
+     */
+    public static SendRequest forSubTxTransition(NetworkParameters params, SubTxTransition subTxTransition) {
+        SendRequest req = new SendRequest();
+        req.tx = new Transaction(params);
+        req.tx.setVersion(3);
+        req.tx.setType(Transaction.Type.TRANSACTION_SUBTX_TRANSITION);
+        req.tx.setExtraPayload(subTxTransition);
```

Comment:
These two lines setting the version and type are redundant because that was added to `Transaction.setExtraPayload`